frontend/src/scenes/insights/insightDataLogic.ts

```diff
@@ -250,19 +256,20 @@ export const insightDataLogic = kea<insightDataLogicType>([
          * that haven't been refactored to use the data exploration yet.
          */
         insightData: (insightData: Record<string, any> | null) => {
-            if (!values.isUsingDataExploration) {
+            if (!values.isUsingDataExploration || insightData === null) {
                 return
             }
 
-            actions.setInsight(
-                {
-                    ...values.insight,
-                    result: insightData?.result,
-                    next: insightData?.next,
-                    filters: values.insight.query ? {} : queryNodeToFilter(values.querySource),
-                },
-                {}
-            )
+            // todo what about results that aren't in this shape?
+            actions.setInsightResults({
+                result: insightData?.result,
+                next: insightData?.next,
+            })
```

Comment:
@thmsobrmlr to check I'm not missing things here ...

spreading values.insight and calling setInsight here was collapsing the inflightFilters and things.

I think all we need to do is update the results we just received back but....

Reply:
At the moment we treat some of the charts as black boxes in the sense that the internals aren't converted yet and still rely on legacy filters. Not setting filters will break these charts unfortunately. An example where this seems to be the case is the retention chart that breaks in this PR.

If this change is really necessary, we could simply convert the query to filters in the chart components themselves. I'd like to avoid converting the internals right now, if possible.

Reply:
Ah, what a pain... good spot.

I had to stop setting the filters because we default to Trends everywhere

so the sequence was running (something like)

1. get insight X with filters Y
2. init all the things for the linked logics
3. they think they are trends because it is not loaded
4. load it
5. on load of data set the filters on the insight *using the current incorrect defaults*

🤔 I think this might turn out to be a pain....

Reply:
![concrete case](https://user-images.githubusercontent.com/984817/225565727-ca3e0221-7831-4919-83b0-78f894ad69c4.gif)

If you create a new insight by clicking on the sidebar, you can change tabs fine.

If you create a new insight by clicking on the dropdown and choosing a query based insight you can't change tab